#VRML_SIM R2023a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/floors/protos/RectangleArena.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.54989780963201 0.3196350793540273 0.7716513558649803 2.2983326280712215
  position 1.6693099730365037 -1.9230333152892003 6.406511622414789
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 3 3
  floorTileSize 0.1 0.1
}
Solid {
  translation -2.1628325696215273e-07 -0.6002303903612703 0.09568359999999997
  rotation -6.033432655567122e-12 -6.597923865694483e-09 1 3.1397637672079943
  children [
    DEF Wall Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0.1 0.1
        }
      }
      geometry Box {
        size 1.1 0.05 0.2
      }
    }
  ]
  name "Wall"
  boundingObject USE Wall
  physics Physics {
  }
}
Solid {
  translation 0.4904419283967939 0.35916018767327995 0.08699268498826683
  rotation -2.947007174608355e-16 -3.4926469021781055e-13 0.9999999999999999 3.139889661090315
  scale 0.901557 0.901557 0.901557
  children [
    DEF Wall Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0.1 0.1
        }
      }
      geometry Box {
        size 1.1 0.05 0.2
      }
    }
  ]
  name "Wall(3)"
  boundingObject USE Wall
  physics Physics {
  }
}
Solid {
  translation -0.5300579899063192 -0.019997310099125774 0.09568390033317792
  rotation -0.0002647159215188264 -0.00026476426967473725 0.9999999299126788 1.570795447957393
  children [
    DEF Wall Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0.1 0.1
        }
      }
      geometry Box {
        size 1.1 0.05 0.2
      }
    }
  ]
  name "Wall(2)"
  boundingObject USE Wall
  physics Physics {
  }
}
Solid {
  translation 0.520001187082124 -0.12192393625495468 0.09717471999999999
  rotation -1.4752691247939097e-15 -1.4989379465285924e-15 1 1.5707947165621832
  children [
    DEF Wall Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0.1 0.1
        }
      }
      geometry Box {
        size 0.9 0.04 0.2
      }
    }
  ]
  name "Wall(1)"
  boundingObject USE Wall
  physics Physics {
  }
}
DEF body Robot {
  translation 0.679684 0.0258729 0.021719
  rotation 0.09063608875220718 -0.9958837420403865 0.0008196068198338832 -3.141503170933623
  children [
    GPS {
      translation 0 0 -0.02
      children [
        DEF gps Shape {
          geometry Cylinder {
            height 0.01
            radius 0.006
          }
        }
      ]
      boundingObject USE gps
      physics Physics {
      }
    }
    DistanceSensor {
      translation -0.02 -0.041 -0.017
      rotation 0 0 -1 2.2252956132800117
      children [
        Solid {
          translation 0.004 0 0
          rotation 0 1 0 1.5708
          children [
            Shape {
              geometry Cone {
                bottomRadius 0.004
                height 0.01
              }
            }
          ]
        }
      ]
      name "DSRightBack"
      numberOfRays 2
      aperture 0.6
    }
    DistanceSensor {
      translation 0.02 -0.041 -0.017
      rotation 0 0 -1 0.9162996938995747
      children [
        Solid {
          translation 0.004 0 0
          rotation 0 1 0 1.5708
          children [
            Shape {
              geometry Cone {
                bottomRadius 0.004
                height 0.01
              }
            }
          ]
        }
      ]
      name "DSLeftBack"
      numberOfRays 2
      aperture 0.6
    }
    DistanceSensor {
      translation -0.044 0.01 -0.017
      rotation 0 0 -1 3.14
      children [
        Solid {
          translation 0.004 0 0
          rotation 0 1 0 1.5708
          children [
            Shape {
              geometry Cone {
                bottomRadius 0.004
                height 0.01
              }
            }
          ]
        }
      ]
      name "DSRightSide"
      numberOfRays 2
      aperture 0.6
    }
    DistanceSensor {
      translation 0.044 0.01 -0.017
      rotation 0 0 -1 0
      children [
        Solid {
          translation 0.004 0 0
          rotation 0 1 0 1.5708
          children [
            Shape {
              geometry Cone {
                bottomRadius 0.004
                height 0.01
              }
            }
          ]
        }
      ]
      name "DSLeftSide"
      numberOfRays 2
      aperture 0.6
    }
    DistanceSensor {
      translation -0.02 0.041 -0.017
      rotation 0 0 -1 -1.9634996938995748
      children [
        Solid {
          translation 0.004 0 0
          rotation 0 1 0 1.5708
          children [
            Shape {
              geometry Cone {
                bottomRadius 0.004
                height 0.01
              }
            }
          ]
        }
      ]
      name "DSRightFront"
      numberOfRays 2
      aperture 0.6
    }
    DistanceSensor {
      translation 0.02 0.041 -0.017
      rotation 0 0 -1 -1.1765
      children [
        Solid {
          translation 0.004 0 0
          rotation 0 1 0 1.5708
          children [
            Shape {
              geometry Cone {
                bottomRadius 0.004
                height 0.01
              }
            }
          ]
        }
      ]
      name "DSLeftFront"
      numberOfRays 2
      aperture 0.6
    }
    Solid {
      translation 0 0 -0.016
      rotation 0 0 1 3.14159
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 0 0
            }
          }
          geometry Cylinder {
            bottom FALSE
            height 0.014
            radius 0.051
            top FALSE
            subdivision 360
          }
        }
      ]
      name "head"
      boundingObject Cylinder {
        height 0.007
        radius 0.05
      }
      physics Physics {
      }
      rotationStep 0.3
    }
    DEF WL HingeJoint {
      jointParameters HingeJointParameters {
        position 1496.579581722081
        anchor 0.05 0 0.002
      }
      device [
        RotationalMotor {
          name "LWM"
        }
      ]
      endPoint Solid {
        translation 0.05 0 0.002
        rotation -0.5212667750912295 -0.6767582491134801 -0.519883853800872 4.339092415437381
        children [
          DEF LW Shape {
            appearance Appearance {
              material Material {
                diffuseColor 1 0 0
              }
            }
            geometry Cylinder {
              height 0.01
              radius 0.02
            }
          }
        ]
        name "solid(1)"
        boundingObject USE LW
        physics Physics {
        }
      }
    }
    DEF WR HingeJoint {
      jointParameters HingeJointParameters {
        position 1653.578239942348
        anchor -0.05 0 0.002
      }
      device [
        RotationalMotor {
          name "RWM"
        }
      ]
      endPoint Solid {
        translation -0.05 0 0.002
        rotation -0.4633745558516352 -0.7553585007009226 -0.4633762579245447 4.435432595828649
        children [
          DEF RW Shape {
            appearance Appearance {
              material Material {
                diffuseColor 1 0 0
              }
            }
            geometry Cylinder {
              height 0.01
              radius 0.02
            }
          }
        ]
        boundingObject USE RW
        physics Physics {
        }
      }
    }
    DEF body Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0.2 0.4 0.2
          emissiveColor 0 0.04 0.1
        }
      }
      geometry DEF body Cylinder {
        height 0.0435
        radius 0.05
      }
    }
  ]
  boundingObject USE body
  physics Physics {
  }
  controller "wall_Fallower"
}
